File: 55-jump-game/jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n= (int)nums.size(), i=0, maxm=0;
        for(i=0;i<min(maxm+1,n); ++i){
            maxm=max(maxm,i+nums[i]);
        }
        return i==n;
    }
};

File: 55-jump-game/jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n= nums.length, curr=0, last=0;
        while(last<=n-2 && curr<=last){
            last= Math.max(last,curr+nums[curr++]);
        }
        return last>=n-1;
    }
}